src/Aether/Devices/Drivers/Sgp4x.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System.Device.I2c;
+
+namespace Aether.Devices.Drivers
+{
+    /// <summary>
+    /// Air Quality and Gas VOC Sensor SGP4x
+    /// </summary>
+    public sealed class Sgp4x : System.IDisposable
+    {
+        /// <summary>
+        /// The default I²C address of this device.
+        /// </summary>
+        public const int DefaultI2cAddress = 0x59;
+
+        private readonly I2cDevice _device;
+
+        /// <summary>
+        /// Instantiates a new <see cref="Sgp4x"/>.
+        /// </summary>
+        /// <param name="device">The I²C device to operate on.</param>
+        public Sgp4x(I2cDevice device)
+        {
+            _device = device;
+            Reset();
+        }
+
+        /// <inheritdoc/>
+        public void Dispose() =>
+            _device.Dispose();
+
+        /// <summary>
+        /// Resets the device.
+        /// </summary>
+        public void Reset()
+        {
+            _device.WriteByte(0x00);
+            _device.WriteByte(0x06);
+            Thread.Sleep(1);
+        }
+
+        /// <summary>
+        /// Gets the serial number of the device.
+        /// </summary>
+        /// <returns>The serial number of the device.</returns>
+        public ReadOnlySpan<byte> GetSerialNumber()
+        {
+            Span<byte> serialNumberWithCRC = stackalloc byte[9];
+
+            // Transmit command
+            _device.WriteByte(0x36);
+            _device.WriteByte(0x82);
+
+            Thread.Sleep(1);
+
+            // Read first two bytes of serial number (+ CRC)
+            Sensirion.ReadUInt16BigEndianAndCRC8(serialNumberWithCRC.Slice(0, 3));
+            
+            // Read second two bytes of serial number (+ CRC)
+            Sensirion.ReadUInt16BigEndianAndCRC8(serialNumberWithCRC.Slice(4, 3));
+            
+            // Read third two bytes of serial number (+ CRC)
+            Sensirion.ReadUInt16BigEndianAndCRC8(serialNumberWithCRC.Slice(7, 3));
+
+            // Read serial number into array excluding CRC bytes
+            byte[] serialNumber = new byte[6];
+
+            serialNumber[0] = serialNumberWithCRC[0];
+            serialNumber[1] = serialNumberWithCRC[1];
+            serialNumber[2] = serialNumberWithCRC[3];
+            serialNumber[3] = serialNumberWithCRC[4];
+            serialNumber[4] = serialNumberWithCRC[6];
+            serialNumber[5] = serialNumberWithCRC[7];
+
+            return new ReadOnlySpan<byte>(serialNumber);
+        }
+
+        /// <summary>
+        /// Runs the sensors self test.
+        /// </summary>
+        /// <returns>True if all tests passed. False if one or more tests failed.</returns>
+        public bool RunSelfTest()
+        {
+            Span<byte> testResultWithCRC = stackalloc byte[3];
+
+            // Transmit command
+            _device.WriteByte(0x28);
+            _device.WriteByte(0x0E);
```

Comment:
This will generate different protocol on the wire than `Write(span)`. Is it correct? In `GetVOCRawMeasure` two command bytes are being written together in one span.